docs/console/design-your-projects/custom-resources/custom-resources.md

```diff
@@ -91,12 +97,48 @@ To delete a Custom Resource, you have to click on the delete button at the botto
 
 You need to insert the Custom Resource name and click on the delete button.
 
-![Delete Custom Resource](./img/custom-resources/delete.png)
+![Delete Custom Resource](./img/delete.png)
+
+## How to manage a Custom Resource in the Marketplace
+
+Users with appropriate permissions can manage the lifecycle of a Custom Resource on the Marketplace, being able to publish their own Custom Resources and update them.
+
+Custom Resources are published to the Marketplace using the `custom-resource` type in the marketplace item schema. Versioning is supported, allowing users to make updates without overwriting previous versions of the Custom Resource.
+
+To learn more on how to add or manage a Custom Resource in the Marketplace, see the [Add Custom Resource to the Marketplace](/marketplace/add_to_marketplace/add_item_by_type/add_custom_resource.md) section.
+
+### Template-based Custom Resource management
+
+:::info
+Please note that template-based Custom Resources are only supported in Projects using the [Enhanced Workflow](/development_suite/set-up-infrastructure/overview.md).
+:::
+
+By default, Custom Resources available in the Marketplace are built using templates. The custom resource item in the Marketplace defines a `generator` field that specifies the template and base folder for generated files. The `apiVersion` field is also set to the special value `custom-generator.console.mia-platform.eu/v1`, necessary to dynamically generate the resource from the Marketplace.
+
+Templates are stored as strings and interpolated at deployment time using the [mustache template system](https://mustache.github.io/). After creating the resource, the user can proceed with its deployment. The template outlines the configuration files that will be generated during deployment: during this process the template is populated with user-provided values, and the resulting files are stored in a dedicated folder in the Project repository, as defined in the Marketplace item specification.
+
+Here is a sequence of what happens when a template-based Custom Resource is deployed:
+
+```mermaid
+sequenceDiagram
+  participant User
+  participant Console
+  User->>Console: Save configuration with template-based Custom Resource<br/>generated from Marketplace
+  Console->>User: Configuration saved
+  User->>Console: Deploy the configuration
+  Console->>Console: Get custom resource from the Marketplace
+  Console->>Console: Interpolate the template with user-defined values
+  Console->>Console: Generate configuration files and store them in a dedicated folder
+  Console->>User: Deployment successful
+```
+
+:::info
+It is important to note that deploying template-based Custom Resources for different runtimes than Kubernetes will require additional configuration on the deployment pipeline.
+:::
```

Comment:
should we add more details here? I think this sentence alone is quite useless without a reference to specific advice or configuration tips